module/mfrc522_user_command.c

```diff
@@ -63,6 +63,38 @@ static int mem_read(char *answer)
 	return byte_amount;
 }
 
+/**
+ * Write data into the MFRC522's internal memory
+ *
+ * @param data User input to write to the memory
+ * @param data_len Length of the data to write
+ *
+ * @return 0 on success, -1 on error
+ */
+static int mem_write(char *data, u8 data_len)
+{
+	u8 remaining_bytes;
+	u8 null_char = '\0';
+
+	if (mfrc522_fifo_write(data, data_len) < 0) {
+		pr_err("[MFRC522] Couldn't write to FIFO\n");
+		return -1;
+	}
+
+	// Complete the remaining slot with NULL bytes
+	for (remaining_bytes = data_len; remaining_bytes < MFRC522_MAX_DATA_LEN;
+	     remaining_bytes++)
+		mfrc522_fifo_write(&null_char, 1);
```

Comment:
This can be simplified:
``` C
static const char[MFCR522_MEM_SIZE] null_byte_array = {0};
mfrc522_fifo_write(null_byte_array, MFRC522_MEM_SIZE - data_len);

```

Reply:
Fixed in 2ed7f94